tests/Makefile.am

```diff
@@ -12,19 +12,23 @@ AM_CPPFLAGS = -I$(top_srcdir) -I$(top_builddir) $(CXX_FLAGS) \
 # -----------------------------------------------------------
 # TESTS declares the tests to actually run;
 # check_PROGRAMS are the binaries to build.
-TESTS = dict-reopen multi-thread
-check_PROGRAMS = dict-reopen multi-thread
+check_PROGRAMS = dict-reopen multi-thread mem-leak
+TESTS = $(check_PROGRAMS)
+
+LDFLAGS += $(LINK_CXXFLAGS)
 
 dict_reopen_SOURCES = dict-reopen.cc
 multi_thread_SOURCES = multi-thread.cc
+mem_leak_SOURCES = mem-leak.cc
 
 LDADD = $(top_builddir)/link-grammar/liblink-grammar.la
 if HAVE_SQLITE
 LDADD += $(SQLITE3_LIBS)
 endif
 
+multi_thread_LDADD = -lpthread $(LDADD)
```

Comment:
Older versions of configure.in had a test for HAVE_PTHREAD   -- I may have removed it, as an attempt to simlify things, as it was mostly unused.  I'm not sure, but perhaps MacOS or WIndows don't have pthreads.